bluepysnap/edges/edges.py

```diff
@@ -46,7 +46,7 @@ def population_names(self):
         """Defines all sorted edge population names from the Circuit."""
         return sorted(self._circuit.to_libsonata.edge_populations)
 
-    def ids(self, group=None, sample=None, limit=None):
+    def ids(self, group=None, sample=None, limit=None, **_):  # pylint: disable=arguments-differ
```

Comment:
How (if at all) can we handle this without disabling `arguments-differ`?

Reply:
Still for discussion, I was also considering another option that wouldn't require to change the signature of ids() and get().
It would be possible to define a class `NodeSet` that can be returned by `node_sets[node_set_name]`, and that can be passed as `group` to ids() and get().

So, instead of doing
```python
sim.circuit.get(name, node_sets=sim.node_sets)
```
you can override the circuit node_sets for example in this way:
```python
sim.circuit.get(sim.node_sets[name])
```
then get() can internally use the NodeSet to select the ids.

However, there would be a limitation, because in this way it's not possible to specify at the same time a custom node_sets and a query dict containing a node set name, for example when the query is:
```python
{"$node_set": "Node12_L6_Y", "node_id": 1}
```

Reply:
yeah, we can do something like
```python
sim.circuit.get({
    "$node_set": "Node12_L6_Y", 
    "$node_sets": sim.node_sets,  # probably better to have a more distinctive name like "$node_sets_instance" 
})
```

Reply:
is that what you were thinking @GianlucaFicarelli? I thought there might be a more clean solution.

Reply:
When filtering by just a single node_set (from a node_sets that's not from the circuit, but from the simulation), I was thinking to:
```python
node_set = sim.node_sets["Node12_L6_Y"]
sim.circuit.get(node_set)
```
If the node_sets come from an external file (not circuit, not simulation), it could be:
```python
node_sets = NodeSets.from_file(...) # or NodeSets(...)
node_set = node_sets["Node12_L6_Y"]
sim.circuit.get(node_set)
```
We can still keep the usual shorter way (when the node_sets is circuit.node_sets):
```python
sim.circuit.get("Node12_L6_Y")
```

If a query is needed when using a different node_set at the same time, something like:
```python
node_set = sim.node_sets["Node12_L6_Y"]
sim.circuit.get({"$node_set": node_set, "morphology": "morph-B")})
```
The downside of this query is that it cannot be dumped as regular json, but maybe this is not desired or required.

About using different keys for node_set name and node_set instance, I'm not convinced that it would make the API easier, but it can be discussed.